From 3ae6b8271533d3cbc42e9f236518c0aa2c583265 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sat, 27 Aug 2022 20:50:43 +0200
Subject: power: supply: axp20x-battery: Enable poweron by RTC alarm

For the Pinephone to be able to poweron on RTC alarm, some bits in
PMIC need to be enabled. This will cause PMIC to be wakeable by
SoC pulling the interrupt line low.

This may need some coordination from a bootloader.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/power/supply/axp20x_battery.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/power/supply/axp20x_battery.c b/drivers/power/supply/axp20x_battery.c
index abdc406a3f46..1cd56e4c1239 100644
--- a/drivers/power/supply/axp20x_battery.c
+++ b/drivers/power/supply/axp20x_battery.c
@@ -1334,6 +1334,11 @@ static int axp20x_power_probe(struct platform_device *pdev)
 		ret = regmap_update_bits(axp20x_batt->regmap, 0x84, 0x37, 0x31);
 		if (ret)
 			goto warn_bat;
+
+		// enable poweron by RTC
+		ret = regmap_update_bits(axp20x_batt->regmap, 0x8f, BIT(7), BIT(7));
+		if (ret)
+			goto warn_bat;
 	}
 
 	return 0;
-- 
2.35.3

